style_guides/architecture.md

```diff
@@ -34,3 +34,36 @@ plugin_root:
   <dd>This folder is where code that is useful on both the client and the server belongs. A consistent example of this is constants, but this could apply to helper modules as well.</dd>
   <dd><strong>NOTE</strong>: If you'd like to avoid adding <code>../common</code> to your public code, you could use <em>webpackShims</em> to resolve the path without traversing backwards.</dd>
 </dl>
+
+### Subdirectories
+
+As a plugin gets more complex, it becomes important to organize your code into subdirectories. Each subdirectory should contain an `index.js` file that exposes the contents of that directory.
+
+```
+plugin_root:
+
+.
+├── common/
+├── public/
+├───── component_one/
+├──────── component_one.js
+├──────── component_one.html
```

Comment:
Do you think it would be helpful to include something like a `component_one_helper.js` in this example as well, as a way of showing that the sub-directories have an additional advantage of keeping helper code that's required only by code in that directory close to it?